metafacture-flux/src/main/java/org/metafacture/flux/HelpPrinter.java

```diff
@@ -113,6 +123,32 @@ private static <T> void describe(final String name, final ObjectFactory<T> facto
 
         final Map<String, Method> attributes = configurableClass.getSetters();
 
+        printAttributes(out, configurableClass, attributes);
+
+        String inString = "<unknown>";
+        String outString = "";
+        final In inClass = moduleClass.getAnnotation(In.class);
+        if (inClass != null) {
+            inString = inClass.value().getSimpleName();
+        }
+        final Out outClass = moduleClass.getAnnotation(Out.class);
+        if (outClass != null) {
+            outString = outClass.value().getSimpleName();
+        }
+        out.println("- signature:\t" + inString + " -> " + outString);
+        if (examplesMap.get(name) != null && examplesMap.get(name).size() == COLUMN_TO_EXAMPLE) {
```

Comment:
It would probably be more future-proof to check for _at least_ the required number of columns.

Also, would it make sense to extract the map value into a variable instead of `get()`ing it all over again?

```suggestion
        final List<String> commandInfo = examplesMap.get(name);
        if (commandInfo != null && commandInfo.size() >= COLUMN_TO_EXAMPLE) {
```

Reply:
Re "extraction of map value": yes of course, thx

Reply:
There is now a variable to check if the reuired columns exist https://github.com/metafacture/metafacture-core/pull/509/commits/3498dc8ceec0250b059153b1f4555e16bbb1b4ef#diff-452d41c10f72d7f32912309d723817ff7424c35c11925c006e803c8d04b37e3bR56

Reply:
I actually preferred the previous, more lenient approach: If the example links doesn't exist then just don't add it; throwing an exception feels a little excessive.

And it's still not future-proof in the sense that it skips the example link in case there happen to be _more_ columns than expected.

So, in conclusion, I stand by my previous suggestion. Please convince me otherwise.

Reply:
> Please convince me otherwise.

I will try:
a) adding links is only started if the file exists - and that's a situation one must deliberatly work on (copy the file to the proper place). Most likely in the event of making a release. In the case a "normal" user starts the `flux.sh` the commands will be printed to stdout, in a good readable way (i.e. not long URLs),  just as before. If so I think it makes sense to bail out when an entry is missing, for every command has the most basic link to the class. (This way I got informed that `encode-yaml` was missing (see https://github.com/metafacture/metafacture-documentation/commit/781caeff015cfb9064b1013307de55b3d7e84b4b)

b) worked on that, see latest commit. Mind that it's allowed to have 2 or 3 columns.

(Didn't amend the commit this time - will squash at the end).

Reply:
Okay, in that case I think it would make (more) sense to validate the entries at load time (`loadExamples()`). And maybe just print a warning for each invalid line.

See 25a2779 for what I have in mind; feel free to ignore/revert.